### PR TITLE
Agent execution script filter (and more)

### PR DESCRIPTION
**Pull request by mprimi**

### CODE REVIEW

**Comment by mprimi** [comment]
Major differences from previous iteration:
* Last 2 commits add then use a new script to filter agent execution (the main purpose of this PR).
* Addressed comments from previous review

New commits with minor changes:
* `Apply test settings to all Gradle Java test configurations`
* `Lower verbosity of UserMetricsTask`

@tgianos

File: genie-web/src/integTest/resources/application-integration.yml [hunk 1]
```diff
@@ -29,6 +29,9 @@ genie:
     cluster-load-balancer:
       source: file:///tmp/genie/loadBalancers/script/source/loadBalance.js
       auto-load-enabled: true
+    execution-mode-filter:
+      source: file:///tmp/genie/loadBalancers/script/source/executionModeFilter.js
```
Comment (tgianos) [inline genie-web/src/integTest/resources/application-integration.yml]: when we change the name of `ClusterLoadBalancer` to `ClusterSelector` we probably want to change these paths too.

File: genie-web/src/main/java/com/netflix/genie/web/scripts/ExecutionModeFilterScript.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+        super(scriptManager, properties, mapper, registry);
+    }
+
+    /**
+     * Evaluate the script and return true if this job should be forced to execute via agent, false if it should be
+     * forced to execute in embedded mode, null if the script decides not explicitly flag this job for one or the other
+     * execution mode.
+     *
+     * @param jobRequest the job request
+     * @return An optional boolean valua
```
Comment (tgianos) [inline genie-web/src/main/java/com/netflix/genie/web/scripts/ExecutionModeFilterScript.java]: typo here for `value`

File: genie-web/src/main/java/com/netflix/genie/web/scripts/ExecutionModeFilterScript.java [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+     * @throws ScriptNotConfiguredException if the script is notyet successfully loaded and compiled
+     * @throws ScriptExecutionException     if the script evaluation produces an error
+     */
+    public Optional<Boolean> forceAgentExecution(
```
Comment (tgianos) [inline genie-web/src/main/java/com/netflix/genie/web/scripts/ExecutionModeFilterScript.java]: Why does it return an `Optional`? Sorry if this is in another part of this PR but wouldn't it just be true/false or exception? Are you using `Optional.empty()` to represent no preference?

File: genie-web/src/test/resources/com/netflix/genie/web/scripts/execution-mode-filter.groovy [hunk 4]
```diff
@@ -0,0 +1,32 @@
+/*
+ *
+ *  Copyright 2017 Netflix, Inc.
```
Comment (tgianos) [inline genie-web/src/test/resources/com/netflix/genie/web/scripts/execution-mode-filter.groovy]: year is off

File: genie-web/src/main/java/com/netflix/genie/web/scripts/ExecutionModeFilterScript.java [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+ * The contract between the script and the Java code is that the script will be supplied global variables
+ * {@code jobRequest} which will be JSON strings representing job request that kicked off this evaluation.
+ * The code expects the script to either return the true (to force agent execution), false (to force legacy/embedded
+ * execution) or null (for no preference).
+ *
+ * @author mprimi
+ * @since 4.0.0
+ */
+@Slf4j
+public class ExecutionModeFilterScript extends ManagedScript {
```
Comment (tgianos) [inline genie-web/src/main/java/com/netflix/genie/web/scripts/ExecutionModeFilterScript.java]: Shouldn't this be implementing some other interface reflecting that it is some sort of `ExecutionModeSelector`? Not sure if we have that for the Headers or Property based paths?

File: genie-web/src/integTest/resources/application-integration.yml (hunk 1 above)
Comment (mprimi) [reply genie-web/src/integTest/resources/application-integration.yml]: Ah yeah, actually not even sure why these are there given that these files won't exist. Test that the app starts and work regardless?

File: genie-web/src/main/java/com/netflix/genie/web/scripts/ExecutionModeFilterScript.java (hunk 3 above)
Comment (mprimi) [reply genie-web/src/main/java/com/netflix/genie/web/scripts/ExecutionModeFilterScript.java]: Yep

> The code expects the script to either return the true (to force agent execution), false (to force legacy/embedded execution) or null (for no preference).

File: genie-web/src/main/java/com/netflix/genie/web/scripts/ExecutionModeFilterScript.java (hunk 5 above)
Comment (mprimi) [reply genie-web/src/main/java/com/netflix/genie/web/scripts/ExecutionModeFilterScript.java]: This is just a script object. 
The execution mode is a different thing and uses this.

**Pull request merged**